plugins/out_kafka/kafka_config.c

```diff
@@ -180,6 +180,16 @@ struct flb_kafka *flb_kafka_conf_create(struct flb_output_instance *ins,
     if (tmp) {
         ctx->gelf_fields.level_key = flb_sds_create(tmp);
     }
+    /* Config: Time_Key_Format */
+    tmp = flb_output_get_property("time_key_format", ins);
+    if(tmp){
+        ctx->time_key_format = flb_strdup(tmp);
```

Comment:
at this level I think would be important to do a dummy validation of the given time format in the configuration, like call a static function that does a strftime(3) and check it return value. Otherwise on runtime it might mess up thinks without noticing it.